### ISSUE TITLE
[SOLVED] Unable to update from 3.10.12 to J4 (or J5)

### ISSUE THREAD

**Issue by PixedBo** [joomla/joomla-cms#42249]
### Steps to reproduce the issue

Setup a 3.10.12 Joomla with these server settings:

![image](https://github.com/joomla/joomla-cms/assets/12549961/d87cf012-8653-4d7b-9522-b44b76dd5e9c)

Go to component -> Joomla Update
Set update options -> Joomla Next

![image](https://github.com/joomla/joomla-cms/assets/12549961/d7e78b41-826f-4426-b9ee-680c894ae511)

Check for updates

### Expected result

Joomla4 update pre-check and possibility to update to J4

### Actual result

"I already have the latest version"

![image](https://github.com/joomla/joomla-cms/assets/12549961/d6277cf5-1e51-4cff-896c-4a9758896732)

### System information (as much as possible)

see above

### Additional comments

I've done some research and i've seen in other discrussion that whene the database shows a prefix (like 5.5.5) some issues may occur. Can this be the case as the database is show as version 5.5.5-10.6.15-MariaDB-cll-lve-log
SSL is present and working and can install extensions from zip or directly from JED without problems.

**Comment by richard67** [comment]
@PixedBo The version prefix "5.5.5-" should not be the problem, it is stripped off before comparing versions, as far as I know.

Can it be that you are using the native mysql client (database type "MySQL" in global configuration in backend or "mysql" in configuration.php)?

If so, please check in backend which database types are available by trying to change the database type in global configuration.

Which choices does the dropdown offer? There should be "MySQLi" (mind the "i" at the end) or "MySQL (PDO)" or both.

If only one of these 2 is available, chose that one, if both are available, chose one of these 2.

Then save and check again for updates and report back the result.

Thanks in advance.

**Comment by PixedBo** [comment]
Hello @richard67 and thanks for your answer.
MySQL type is already MySQLi:

![image](https://github.com/joomla/joomla-cms/assets/12549961/82e3d738-11d1-42fc-9d7d-f422010ce1ef)

I have those options available:

![image](https://github.com/joomla/joomla-cms/assets/12549961/634b03b3-ebbb-4eed-b5fd-cb03ec567104)

I will try MySQL (PDO) as soo as i retrieve the FTP credentials in case something goes wrong and can't access the backend anymore, so i will need to restore the configuration.php manually

**Comment by richard67** [comment]
@PixedBo MySQLi should be ok.

**Comment by richard67** [comment]
@PixedBo If you go to "Extensions -> Manage -> Update Sites" in backend, which URL does it show for the "Joomla! Core" update site? Does it show `https://update.joomla.org/core/sts/list_sts.xml`? Or something else?

**Comment by PixedBo** [comment]
Nice! It's disabled and really don't know why :D

![Cattura](https://github.com/joomla/joomla-cms/assets/12549961/4751594a-88f3-4b66-a77d-a3b5e39536b0)

I've enabled it and now it's working,
Thanks again

**Comment by richard67** [comment]
@PixedBo When an update site times out on connection, e.g. because it is not available, then the update site is disabled automatically. It seems that has happened in past, maybe when we had problems with the update servers on October 17 when we had released 4.4.0 and 5.0.0. So it's always a good idea to check the update sites from time to time.

Closing as solved by enabling the update site again.

Thanks for reporting back.
